File: Inceptezpyspark-27/Sparkcore/Lab10_leftjoin.py
from pyspark import SparkContext

def main():
    
    #sc = SparkContext(master = "local", appName = "Lab10")
    sc = SparkContext(appName = "lab10")
    sc.setLogLevel("ERROR")
    
    emp = sc.textFile("file:/home/hduser/sparkdata/emp.csv").map(lambda x : x.split(",")).map(lambda x : (x[2],(x[0],x[1],x[3])))
    
    emp.foreach(print)
    
    dept = sc.textFile("file:/home/hduser/sparkdata/dept.csv").map(lambda x : x.split(",")).map(lambda x : (x[0],x[1]))
        
    print("==============================================")
    
    dept.foreach(print)
    
    empdept = emp.leftOuterJoin(dept)
    
    print("==============================================join")
    
    empdept.foreach(print)
    
main()

File: InceptezSpark-27/src/inceptez/sparkcore/Lab10_leftjoin.scala
package inceptez.sparkcore
import org.apache.spark.SparkContext

object Lab10_leftjoin {
  
  def main(args: Array[String])=
  {
    //val sc = new SparkContext(master = "local", appName = "Lab10")
    val sc = new SparkContext
    sc.setLogLevel("ERROR")
    
    val emp = sc.textFile("file:/home/hduser/sparkdata/emp.csv")
                .map(x => x.split(","))
                .map(x => (x(2),(x(0),x(1),x(3))))
                
    emp.foreach(println)
    
    val dept = sc.textFile("file:/home/hduser/sparkdata/dept.csv")
                 .map(x => x.split(",")).map(x => (x(0),x(1)))
                 
    
    println("==============================================")
    
    dept.foreach(println)
    
    val empdept = emp.leftOuterJoin(dept)
    
    
    println("==============================================join")
    
    empdept.foreach(println)
    
    
  }
  
  
  
}